text/0000-sbom.md

```diff
@@ -0,0 +1,150 @@
+
+Having a structured and consistent SBOM that can be consumed by downstream systems could greatly enhance the security proposition around buildpacks.
+
+## What use cases does it support?
+
+- Consistent SBOM across buildpacks
+- Improved integration with other supply chain analysis and scanning tools
+
+## What is the expected outcome?
+
+- Use of CycloneDX and SPDX as the SBOM storage format and updates the the specification and lifecycle to reflect the same.
+- Moving the SBOM from `<layer>.toml`, `launch.toml` and `build.toml` to separate files.
+
+## Why support multiple SBOM formats?
+
+The motivation for supporting multiple SBOM formats is because there is currently no clear leader in the SBOM space. SPDX and CycloneDX both specialize in different use cases - compliance and supply chain security respectively, and although both do have a lot of overlap, there is still loss of information when converting between the two. The other motivation is that the SBOM story doesn't end with their generation. Different users could have different SBOM consumers who support one format or the other. Leaving individual SBOM files intact allows users to post-process it later if needed without any loss in information. This allows users to flexibly use the Buildpacks API to generate a SBOM in the format that is most useful to them.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Currently the SBOM is defined in the `<layer>.toml`, `launch.toml` and `build.toml` respectively under the `[bom]` table.
+It may not be the most user-friendly way for buildpack authors to create SBOM documents in the above format. This RFC proposes that the `bom` be moved to `<layer>.bom.<ext>`, `launch.bom.<ext>` and `build.bom.<ext>` instead where `<ext>` will be `cdx.json` for CycloneDX documents and `spdx.json` for SPDX documents. A buildpack can also output the `bom` in multiple formats i.e. both CycloneDX and SPDX. 
+
+Additionally the buildpack must also indicate the SBOM format(s) it exports using the in its `buildapck.toml`'s `sbom` key in the `buildpacks` table.
+
+The format of this key should look like - 
+
+```toml
+api = "0.x"
+
+[buildpack]
+id = "<buildpack ID>"
+name = "<buildpack name>"
+# This can be an array of supported SBOM formats by the buildpack.
+# Valid array values are "CycloneDX" and "SPDX" as of this RFC.
+sbom = ["CycloneDX"]
+```
+
+This information can be displayed on the registry and `pack inspect-buildpack`.
+
+The `lifecycle` binaries would be responsible for taking all of the above `bom` files, and copying them in `/layers/config/sbom/<type>/<buildpack-id>/<layer-id>/bom.<ext>` for the layer `bom` files where `type` could be `build` or `launch`. Buildpack level BOM files will reside at `/layers/config/sbom/<type>/<buildpack-id>/bom.<ext>` where the type could be `launch` and `build` respectively.
```

Comment:
/queue-issue buildpacks/lifecycle "Builder should copy bom files to /layers/config/sbom" type/enhancement epic/sbom